Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -910,6 +910,7 @@ JSValue Interpreter::executeProgram(const SourceCode& source, JSGlobalObject*, J
                         return throwException(globalObject, throwScope, createTDZError(globalObject));
                     baseObject = scope;
                 }
+                RETURN_IF_EXCEPTION(throwScope, JSValue());
```

Comment:
The better way to this is:
```
bool hasProperty = scope->hasProperty(globalObject, ident));
RETURN_IF_EXCEPTION(throwScope, JSValue());
if (hasProperty) {
    ...
```

Reply:
I see that Alexey just suggested the same thing.

Reply:
Updated.